packages/hash/api/src/model/index.ts

```diff
@@ -44,6 +44,9 @@ export { default as UserModel } from "./knowledge/user.model";
 export * from "./knowledge/org.model";
 export { default as OrgModel } from "./knowledge/org.model";
 
+export * from "./knowledge/org.model";
+export { default as OrgMembershipModel } from "./knowledge/orgMembership.model";
```

Comment:
Can you explain to me why we do this?
- We export `default` as `OrgMembershipModel` from the module, where we define it
- We import `OrgMembershipModel` from the module where we just exported *the same* class from

Sorry, but this absolutely makes no sense to me. Why we don't `export class OrgMembershipModel` in `orgMembership.model.ts`?

This is very error-prone and also completely breaks IntelliSense. Error example: in *entity-type.model.ts* we use 
```js
import {
  EntityTypeModel,
  PropertyTypeModel,
  LinkTypeModel,
  UserModel,
  DataTypeModel,
} from "../index";
import dataTypeModel from "./data-type.model";
import linkTypeModel from "./link-type.model";
```
We import `dataTypeModel` and `linkTypeModel`, which are the exact same classes as `DataTypeModel` and `LinkTypeModel`, both imported from different files with different names.

This, btw, was the first file I looked at to understand the pattern.

Getting into this state is pretty easy: when typing `linktypemodel`, VSCode suggests to import `linkTypeModel` from `./link-type.model` over `LinkTypeModel`. It feels like importing by strings, not by identifiers.

We use this pattern all over the place (which is very concerning) so this shouldn't block merging the PR but we should get to this ASAP.

My proposal is that we shouldn't use `export default` **at all**.

Reply:
I quickly searched the web and it seems that I'm not the only one who finds this concerning, I have not seen a single recommendation to use `export default` but to avoid it.

Reply:
This is to do with circular imports. @benwerner01 maybe don't spend more time explaining it here, we should find one place to explain this well.

Reply:
[Moved to slack](https://hashintel.slack.com/archives/C022217GAHF/p1663174353728269) (_internal_)